agent/lib/kontena/websocket_client.rb

```diff
@@ -166,12 +179,15 @@ def on_close(event)
       @connected = false
       @connecting = false
       @ws = nil
-      if event.code == 4001
+
+      case event.code
+      when 4001
         handle_invalid_token
-      elsif event.code == 4010
-        handle_invalid_version
+      when 4010
+        handle_invalid_version(event.reason)
+      else
```

Comment:
What other codes should be handled specially? Should the agent abort on 404x close events as well (conflicts on the server `HostNode` state)? It should definitely just reconnect on the 4030 ping timeout, though.